### PR TITLE
Improve DEVELOP.rst

### PR DESCRIPTION
**Pull request by amotl**
Hi there,

this is an overhaul of "DEVELOP.rst". The patch brings in a better document structure and a more concise walkthrough in order to get started in development mode more quickly.

With kind regards,
Andreas.

### CODE REVIEW

File: DEVELOP.rst [hunk 1]
```diff
@@ -1,68 +1,72 @@
...
 
 
-Building the App
-================
+*****
+Tests
+*****
 
-Build the app like so::
+You can run the tests like::
```
Comment (justien) [inline DEVELOP.rst:27]:
```suggestion
To run the tests, use the following commands::
```

File: DEVELOP.rst [hunk 2]
```diff
@@ -1,68 +1,72 @@
...
+    npm run cratedb
+
+Start the development server on port ``9000``::
 
+    npm run develop
 
-Running CrateDB
-===============
+After the application has been compiled and the Webpack development server was
+started, the process should automatically open a browser tab.
```
Comment (justien) [inline DEVELOP.rst:49]:
```suggestion
After the application has been compiled and the Webpack development server has
started, the process should automatically open a browser tab.
```

File: DEVELOP.rst [hunk 3]
```diff
@@ -1,68 +1,72 @@
...
 You will need to enable *Cross-Origin Resource Sharing* in your `CrateDB
 configuration`_ to test the Admin UI against it::
 
     http.cors.enabled: true
     http.cors.allow-origin: "*"
 
+The quickest way to spin up an instance like that is probably to use Docker::
+
+    docker run -it --rm --publish 4200:4200 crate/crate:nightly -Chttp.cors.enabled=true -Chttp.cors.allow-origin=*
+
```
Comment (justien) [inline DEVELOP.rst:69]:
```suggestion
A quick way to spin up an instance of this is to use Docker::

    docker run -it --rm --publish 4200:4200 crate/crate:nightly -Chttp.cors.enabled=true -Chttp.cors.allow-origin=*

```

File: DEVELOP.rst [hunk 4]
```diff
@@ -71,10 +75,14 @@ node, like so::
...
-Adding New Dependencies
+**********************
+Development operations
+**********************
+
+Adding new dependencies
 =======================
 
-Application dependencies should be added as follow::
+In order to add application dependencies, just invoke::
```
Comment (justien) [inline DEVELOP.rst:85]: I recommend against using "just", as it does have a slightly patronising tone.

Try:

"To add application dependencies, invoke::"

File: DEVELOP.rst [hunk 5]
```diff
@@ -84,33 +92,25 @@ for the application to load correctly.
 If the dependency is an angular module, it should also be included in ``MODULES``
 in ``./app/app.module.js``
 
-Dev dependencies should be added as follow::
+In order to add development dependencies, just invoke::
```
Comment (justien) [inline DEVELOP.rst:95]:
```suggestion
To add development dependencies, invoke::
```

File: DEVELOP.rst [hunk 6]
```diff
@@ -120,7 +120,23 @@ On startup, the ``conf/plugins.json`` file is read and plugins
 The `tutorial plugin`_ is loaded by the default configuration.
 
 
-Preparing a Release
+********
+Building
+********
+
+In order to compile the app without development mode, just invoke::
```
Comment (justien) [inline DEVELOP.rst:127]:
```suggestion
To compile the app without development mode, invoke::
```

**Review by justien — CHANGES_REQUESTED** [review]
Minor tone and style changes: otherwise looks great!

**Comment by amotl** [comment]
Hi Justine,

thank you for your extensive review. I applied all of your excellent suggestions without further ado.

With kind regards,
Andreas.

File: DEVELOP.rst [hunk 7]
```diff
@@ -1,68 +1,72 @@
...
 You will need to enable *Cross-Origin Resource Sharing* in your `CrateDB
 configuration`_ to test the Admin UI against it::
 
     http.cors.enabled: true
     http.cors.allow-origin: "*"
 
+A quick way to spin up an instance of this is to use Docker::
+
+    docker run -it --rm --publish 4200:4200 crate/crate:nightly -Chttp.cors.enabled=true -Chttp.cors.allow-origin=*
+
```
Comment (seut) [inline DEVELOP.rst:69]: This section confuses me. The above sections tells me that I can/should start crate by `npm run cratedb`.
So its already started including the cors config.
I guess this section should server as an alternative to `npm run cratedb`, so we should definitly highlight it as as alternative ;)

Comment (seut) [reply DEVELOP.rst:69]: Or maybe delete it completely and just talk about the `base_uri` in order to connect to different instances

Comment (amotl) [reply DEVELOP.rst]: > I guess this section should server as an alternative to `npm run cratedb`.

That was its original intention, right.

> Or maybe delete it completely and just talk about the `base_uri` in order to connect to different instances.

You are right. I amended this again on behalf of 87c99096c.

**Review by seut — APPROVED** [review]
👍

**Pull request merged**